Modules/EMCAL/src/PedestalChannelCheck.cxx

```diff
@@ -0,0 +1,202 @@
+        double mean, sigma;
+        meanfinder.EvaluateUni(smcounts.size(), smcounts.data(), mean, sigma);
+        int outside_counts = 0.0;
+        if (mean < mBadPedestalChannelADCRangeLow || mean > mBadPedestalChannelADCRangeHigh) {
+          result = Quality::Bad;
+          messagebuilder << "Error: Average Pedestal outside limits (" << mBadPedestalChannelADCRangeLow << "-" << mBadPedestalChannelADCRangeHigh << ") " << std::endl;
+        } else if (mean < mMedPedestalChannelADCRangeLow || mean > mMedPedestalChannelADCRangeHigh) {
+          result = Quality::Medium;
+          messagebuilder << "Warning: Average Pedestal outside limits (" << mMedPedestalChannelADCRangeLow << "-" << mMedPedestalChannelADCRangeHigh << ") " << std::endl;
+        }
+
+        for (auto ib : ROOT::TSeqI(0, h->GetXaxis()->GetNbins())) {
+          if (h->GetBinContent(ib + 1) > (mean + mSigmaPedestalChannel * sigma)) {
+            outside_counts += 1;
+          }
+        }
+        Float_t out_counts_frac = (float)(outside_counts) / (float)(h->GetXaxis()->GetNbins());
+
+        if (out_counts_frac > mBadPedestalChannelOverMeanThreshold) {
+          result = Quality::Bad;
+          messagebuilder << "Error: Fraction of Outliers Above " << mMedPedestalChannelOverMeanThreshold << std::endl;
+        } else if (out_counts_frac > mMedPedestalChannelOverMeanThreshold) {
+          result = Quality::Medium;
+          messagebuilder << "Warning: Fraction of Outliers Above " << mMedPedestalChannelOverMeanThreshold << std::endl;
+        }
+      }
+    }
+  }
+
+  result.addFlag(o2::quality_control::FlagTypeFactory::BadEMCalorimetry(), messagebuilder.str());
+
+  return result;
+}
+
+std::string PedestalChannelCheck::getAcceptedType() { return "TH1"; }
+
+void PedestalChannelCheck::beautify(std::shared_ptr<MonitorObject> mo, Quality checkResult)
+{
+  std::array<std::string, 4> channelHists = { { "mPedestalChannelFECHG", "mPedestalChannelFECLG", "mPedestalChannelLEDMONHG", "mPedestalChannelLEDMONLG" } };
+  if (std::find(channelHists.begin(), channelHists.end(), mo->getName()) != channelHists.end()) {
+    auto* h = dynamic_cast<TH1*>(mo->getObject());
+    TPaveText* msg = new TPaveText(0.12, 0.84, 0.88, 0.94, "NDC");
+    h->GetListOfFunctions()->Add(msg);
+    msg->SetName(Form("%s_msg", mo->GetName()));
+    if (checkResult == Quality::Good) {
+      msg->Clear();
+      msg->AddText("Data OK: Average Pedestal & Fraction of Outliers Within Limits");
+      msg->SetFillColor(kGreen);
+      msg->Draw();
+    } else if (checkResult == Quality::Bad) {
+      ILOG(Debug, Devel) << "Quality::Bad, setting to red";
+      stringstream msg_bad;
+      msg_bad << "Error: Average Pedestal outside limits (" << mBadPedestalChannelADCRangeLow << "-" << mBadPedestalChannelADCRangeHigh << ") or Fraction of Outliers Above " << mBadPedestalChannelOverMeanThreshold << std::endl;
+      msg->Clear();
+      msg->AddText(msg_bad.str().c_str());
```

Comment:
You should make use of the flag you defined in the check function. This allows to provide more meaningful messages. Please change in next iteration. As example see https://github.com/AliceO2Group/QualityControl/blob/master/Modules/EMCAL/src/RawCheck.cxx (beautify function)